finetune/saver.py

```diff
@@ -129,21 +142,44 @@ def load(self, path):
 
     def get_scaffold_init_fn(self):
         
-        def init_fn(scaffold, session):
+        def init_fn(scaffold, session, model_portion=None):
+            self.var_val = []
             if self.variables is not None:
                 variables_sv = self.variables
             else:
                 variables_sv = dict()
             all_vars = tf.global_variables()
-            self.var_val = []
-            for var in all_vars:
-                for saved_var_name, saved_var in itertools.chain(variables_sv.items(), self.fallback.items()):
-                    if saved_var_name == var.name:
-                        for func in self.variable_transforms:
-                            saved_var = func(var.name, saved_var)
-                        var.load(saved_var, session)
-                        break
-                            
+            is_dict = False
+            if model_portion != 'entire_model': #we must be loading in the case of two separate estimators
+                print("loading variables for")
+                print(model_portion)
+                is_dict = True
+                assert model_portion in ['featurizer','target'], "Must be using separate estimators if loading before graph creation"
+                trainables = variables_sv
+                if model_portion == 'featurizer':
+                    norm_variable_scopes = ['b:0', 'g:0']
```

Comment:
This list and the same logic exists twice. I would suggest it might be best to manage these things by adding variables to collections.

Reply:
Also would make sense just to break this up into a few basic functions.  

e.g. `variables_to_load(self, vars, model_portion, refresh_base_model)`
and `load_variables(self, vars, reset_adapters)`
and 

Would also be nice to settle on one of `reset` / `refresh`

Reply:
@madisonmay re: reset/refresh - I've chosen to keep them distinct because they serve different purposes. Refresh is for weight loading, and reset was for removing any previously loaded adapter weights. I've changed 'reset_adapters' to 'zero_out_adapters' to make the distinction more clear